editor/src/components/canvas/canvas-strategies/strategies/set-padding-strategy.tsx

```diff
@@ -398,3 +411,49 @@ function indicatorPosition(
       assertNever(edge)
   }
 }
+
+function opposite(padding: CSSPaddingKey): CSSPaddingKey {
+  switch (padding) {
+    case 'paddingBottom':
+      return 'paddingTop'
+    case 'paddingTop':
+      return 'paddingBottom'
+    case 'paddingLeft':
+      return 'paddingRight'
+    case 'paddingRight':
+      return 'paddingLeft'
+    default:
+      assertNever(padding)
+  }
+}
+
+function adjustPaddings(
+  adjustMode: PaddingAdjustMode,
+  paddingPropInteractedWith: CSSPaddingKey,
+  delta: number,
+  padding: CSSPaddingMappedValues<CSSNumberWithRenderedValue | undefined>,
+  precision: AdjustPrecision,
+): CSSPaddingMappedValues<CSSNumberWithRenderedValue | undefined> {
+  const newPadding = offsetPaddingByEdge(paddingPropInteractedWith, delta, padding, precision)
+  if (adjustMode === 'individual') {
+    return newPadding
+  }
+
+  const newPaddingValue = newPadding[paddingPropInteractedWith]
+
+  if (adjustMode === 'all') {
+    return {
+      paddingTop: newPaddingValue,
+      paddingBottom: newPaddingValue,
+      paddingLeft: newPaddingValue,
+      paddingRight: newPaddingValue,
+    }
+  }
+
+  if (adjustMode === 'cross-axis') {
+    newPadding[opposite(paddingPropInteractedWith)] = newPaddingValue
+    return newPadding
+  }
+
+  assertNever(adjustMode)
```

Comment:
Can you change this to use a `switch` statement please, so that we can clearly see all cases laid out

Reply:
refactored in https://github.com/concrete-utopia/utopia/pull/3210/commits/bee357129367cbbf95dd4ed87b940765c780d51a